crypto/whrlpool/build.info

```diff
@@ -17,13 +17,21 @@ IF[{- !$disabled{asm} -}]
   ENDIF
 ENDIF
 
-SOURCE[../../libcrypto]=wp_dgst.c $WPASM
-
 # Implementations are now spread across several libraries, so the defines
 # need to be applied to all affected libraries and modules.
-DEFINE[../../libcrypto]=$WPDEF
 DEFINE[../../providers/libimplementations.a]=$WPDEF
 
+SOURCE[../../libcrypto]=wp_dgst.c $WPASM
+DEFINE[../../libcrypto]=$WPDEF
```

Comment:
Should this actually be in an "ELSE" of the below "IF"? In the event of a "no-deprecated" build, does anything in libcrypto actually need this code? Or is it *only* used by the legacy provider?

Reply:
I actually tried that first, but it turns out that the WHIRLPOOL functions are also referred to from `crypto/evp/legacy_wp.c`, so the functionality must remain internally in libcrypto as well until the legacy stuff is *gone*.

Reply:
ok